snapcraft.yaml

```diff
@@ -1382,9 +1381,20 @@ parts:
         sed -i 's#libdir=/usr#libdir=${prefix}#' $PC
         sed -i 's#includedir=/usr#includedir=${prefix}#' $PC
       done
-
+  libicuc:
+    source: https://github.com/unicode-org/icu.git
+    source-tag: release-73-1
+    source-subdir: icu4c/source
+    plugin: autotools
+    autotools-configure-parameters:
+      # WORKAROUND: Install to /usr instead of /usr/local because it's not in search paths
+      - --prefix=/usr
+      - --disable-renaming
```

Comment:
Can you test to add `- --libdir=\${exec_prefix}/lib/$CRAFT_ARCH_TRIPLET` to install libraries into /usr/lib/x86_64-... and check that it does work as expected?

Reply:
Actually I was getting this error while building it, but I guess this will fix it.
```
2023-05-02T16:28:02.2492240Z :: Extracting symbols from /root/prime/usr/lib/x86_64-linux-gnu/perl/5.34.0/auto/Compress/Raw/Zlib/Zlib.so into usr/lib/debug/.build-id/7b/ef161cb7a5d54796c8e6028ac11322ab3ce77f.debug
2023-05-02T16:28:02.2501497Z :: Stripping with --strip-unneeded  /root/prime/usr/lib/x86_64-linux-gnu/perl/5.34.0/auto/Compress/Raw/Zlib/Zlib.so
2023-05-02T16:28:02.2510312Z :: Extracting symbols from /root/prime/usr/lib/x86_64-linux-gnu/perl/5.34.0/auto/Math/BigInt/FastCalc/FastCalc.so into usr/lib/debug/.build-id/5b/2317dbb6d1135167696d33ef8d4c44740b3d46.debug
2023-05-02T16:28:02.2517444Z :: Stripping with --strip-unneeded  /root/prime/usr/lib/x86_64-linux-gnu/perl/5.34.0/auto/Math/BigInt/FastCalc/FastCalc.so
2023-05-02T16:28:02.2525001Z :: Extracting symbols from /root/prime/usr/lib/x86_64-linux-gnu/perl/5.34.0/auto/Filter/Util/Call/Call.so into usr/lib/debug/.build-id/d0/1cfc939a6f47ccde67102d04897f728bd212cb.debug
2023-05-02T16:28:02.2531375Z :: Stripping with --strip-unneeded  /root/prime/usr/lib/x86_64-linux-gnu/perl/5.34.0/auto/Filter/Util/Call/Call.so
2023-05-02T16:28:02.2538354Z :: Extracting symbols from /root/prime/usr/lib/x86_64-linux-gnu/perl/5.34.0/auto/Hash/Util/FieldHash/FieldHash.so into usr/lib/debug/.build-id/77/ca84b68b81a6c4d8434953d58dae2805a69750.debug
2023-05-02T16:28:02.2544160Z :: Stripping with --strip-unneeded  /root/prime/usr/lib/x86_64-linux-gnu/perl/5.34.0/auto/Hash/Util/FieldHash/FieldHash.so
2023-05-02T16:28:02.2554149Z :: Traceback (most recent call last):
2023-05-02T16:28:02.2554893Z ::   File "/usr/local/lib/python3.10/dist-packages/craft_parts/sources/git_source.py", line 78, in generate_version
2023-05-02T16:28:02.2566662Z ::     subprocess.check_output(
2023-05-02T16:28:02.2567044Z ::   File "/usr/lib/python3.10/subprocess.py", line 420, in check_output
2023-05-02T16:28:02.2567480Z ::     return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
2023-05-02T16:28:02.2567891Z ::   File "/usr/lib/python3.10/subprocess.py", line 524, in run
2023-05-02T16:28:02.2568271Z ::     raise CalledProcessError(retcode, process.args,
2023-05-02T16:28:02.2580916Z :: subprocess.CalledProcessError: Command '['git', '-C', '/data', 'describe', '--dirty']' returned non-zero exit status 128.
2023-05-02T16:28:02.2581351Z ::
2023-05-02T16:28:02.2582172Z :: The above exception was the direct cause of the following exception:
2023-05-02T16:28:02.2582471Z ::
2023-05-02T16:28:02.2582722Z :: Traceback (most recent call last):
2023-05-02T16:28:02.2590350Z ::   File "/data/strip_binaries.py", line 95, in <module>
2023-05-02T16:28:02.2591033Z ::     version_number = GitSource.generate_version(part_src_dir=os.environ['CRAFT_PROJECT_DIR'])
2023-05-02T16:28:02.2591775Z ::   File "/usr/local/lib/python3.10/dist-packages/craft_parts/sources/git_source.py", line 97, in generate_version
2023-05-02T16:28:02.2599412Z ::     raise errors.VCSError(message=stderr.decode(encoding).strip()) from err
2023-05-02T16:28:02.2600182Z :: craft_parts.sources.errors.VCSError: fatal: detected dubious ownership in repository at '/data'
2023-05-02T16:28:02.2600620Z :: To add an exception for this directory, call:
2023-05-02T16:28:02.2607463Z ::
2023-05-02T16:28:02.2607951Z :: 	git config --global --add safe.directory /data
2023-05-02T16:28:02.4308516Z 'override-prime' in part 'cleanup' failed with code 1.
2023-05-02T16:28:02.4309426Z Review the scriptlet and make sure it's correct.
```
I first thought, that may be due to building this in github made this issue.

Reply:
Wow... that looks like a bug in the debug symbols extracting tool...

Reply:
Mmm... try running `git config --global --add safe.directory /data` in your folder...

Reply:
Still I am building these on github action... I can't.... How can I test them in a local machine😭😭😭😭, It'll take days. Let me see what happens, now.

Reply:
Oh... are you building it inside github????

You should be able to build it using "snapcraft", as usual, in your local machine. But it takes about half an hour or so...

Reply:
Or am I missing something?

Reply:
> Or am I missing something?

With my 10Mb/s speed, it takes me hours

Reply:
It's not only the 10Mb/s; downloading is only a little part of the work. Compiling all those parts use A LOT of time, unfortunately. It would happen to you even with a 1Gb/s link, I'm afraid...

Reply:
I got it built up. I added the `git config --global --add safe.directory /data` before the python line. Check the actions, you'll get artifact.
https://github.com/soumyaDghosh/gnome-sdk/actions/runs/4881611247